react-renderer/src/inspector/views/Mod.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from "react";
+import IViewProps from "./IViewProps";
+import AttrPicker from "./mod/AttrPicker";
+import { staticMap } from "src/componentMap";
+import styled from "styled-components";
+import defmap from "./mod/defmap";
+
+const ShapeItem = styled.li<any>`
+  display: block;
+  padding: 1em;
+  margin-top: -1px;
+  border: 1px solid #d1d1d1;
+  background-color: ${({ selected }: any) =>
+    selected ? "#F9F9F9" : "#f0f0f0"};
+  color: rgba(0, 0, 0, 0.5);
+  font-family: monospace;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  cursor: pointer;
+`;
+
+interface IState {
+  selectedShape: number;
+}
+
+class Mod extends React.Component<IViewProps, IState> {
+  public readonly state = { selectedShape: -1};
+  public setSelectedShape = (key: number) => {
+    this.setState({ selectedShape: key});
+  };
+  public modAttr = (attrname: string, attrval: Value<any>) => { 
+    const {frame, modShapes, frameIndex, history} = this.props;
+    const {selectedShape} = this.state;
+    if (frameIndex === -1 || frameIndex === history.length - 1) { // make it only work on last frame
+      const shap = frame!.shapes[(selectedShape === -1 ? 0 : selectedShape)]; // the -1 index behaves weirdly so workaround
+      if (shap.properties.hasOwnProperty(attrname)) {
+        shap.properties[attrname] = attrval;
+      }
+      else throw new Error("Shape does not have property " + attrname + " .");
+      if (modShapes) modShapes(frame!);
+    }
+  }
+  public render() {
+    const { frame } = this.props;
+    if (frame === null) {
+      return <div />;
+    }
+
+    const { selectedShape } = this.state;
+    return (
+      <div
+        style={{
+          display: "flex",
+          width: "100%",
+          height: "100%",
+          overflow: "hidden",
+        }}
+      >
+        <div
+          style={{ overflowY: "auto", height: "100%", }}
+        >
+          <ul
+            style={{
+              listStyleType: "none",
+              padding: "0 0 1em 0",
+              margin: 0,
+              top: 0,
+              width: 156,
+              left: 0,
+              right: 0,
+            }}
+          >
+            {frame.shapes.map(
```

Comment:
This looks like a copy-paste of code from `ShapeView` to calculate the bbox of a shape. Can you factor it out into `Util.tsx` and use it in both files?

Reply:
Fixed in 0e5d3a803d6d199268633fdf5d6c1d38a92acaea